experiments/test_helpers.rb

```diff
@@ -97,21 +119,21 @@ def start_request_threads
         @num_threads.times do
           @request_threads << Thread.new do
             thread_id = Thread.current.object_id
-            thread_resource = ExperimentalResource.new(
-              name: @resource_name,
-              service: @service,
-              semian: @semian_config,
-            )
-
             @thread_timings[thread_id] = { samples: [] }
 
             sleep_interval = 1.0 / @requests_per_second_per_thread
             until @done
               sleep(sleep_interval)
+              service = @services.sample
+              thread_resource = ExperimentalResource.new(
+                name: "#{@resource_name}_#{service.object_id}",
```

Comment:
make sure that each service has its own resource. Later one, we should have each thread have its own resource as well

Reply:
hmmmm I think you're right. This is wrong. I wanted to retrieve the existing resources, but I think I messed up. Let me fix that